### PR TITLE
feat: Use pythia8 and HepMC from the standard repository

### PR DESCRIPTION
**Pull request by andriish**
Use pythia8 and HepMC from the standard repository. There are no reasons *not* to use them, as the maintainer fo those packages for RedHAt/Fedora is our colleague and member of HepMC team, i.e. those are de-facto "official" binaries.

The only difference is that `pythia8-config` is explicitly removed for RedHat/Fedora... But Geant4 does not rely on it anyway.

Tag @etiennedreyer @pariRieck

### CODE REVIEW

**Comment by etiennedreyer** [comment]
Hi @andriish thanks for the contribution. We need to check if the docker image builds successfully... I'll try submitting the job.

**Comment by etiennedreyer** [comment]
I didn't manage to run the docker build action without having a branch on this repository. So I made a branch `b_test_docker` and am currently trying it out there: https://github.com/cocoa-hep/cocoa-hep/actions/runs/6959595118

**Comment by etiennedreyer** [comment]
> I didn't manage to run the docker build action without having a branch on this repository. So I made a branch `b_test_docker` and am currently trying it out there: https://github.com/cocoa-hep/cocoa-hep/actions/runs/6959595118

Update: the docker build failed with the following error:

```
#21 273.0 Error downloading packages:
#21 273.0   gcc-4.8.5-44.el7.x86_64: [Errno 256] No more mirrors to try.
#21 273.0 
#21 ERROR: process "/bin/sh -c yum -y install root &&     yum clean all" did not complete successfully: exit code: 1
```
However, I might have done something silly, so I'd appreciate @andriish if you could have a look 🕵️

**Comment by etiennedreyer** [comment]
Hi @andriish I tried it again and unfortunately it failed the same way:
https://github.com/cocoa-hep/cocoa-hep/actions/runs/6959595118/job/18969515810
Do you have any idea if there might be something misconfigured in the docker build workflow? We did successfully run this action in previous versions of the main branch. Thanks!

**Comment by andriish** [comment]
Hi @etiennedreyer, what I see now is that there are multiple problems:
- the problem with repository you see. Here one can do very little. Maybe just use some other location of that file or 
something like this: https://unix.stackexchange.com/questions/444651/how-to-add-a-centos-repo-having-url-of-packages
 - for some reasons the checkout action does not work and always picks up just one particular commit.
See e.g. https://github.com/cocoa-hep/cocoa-hep/pull/45/checks
```
Getting Git version info
Deleting the contents of '/__w/cocoa-hep/cocoa-hep'
The repository will be downloaded using the GitHub REST API
To create a local Git repository instead, add Git 2.18 or higher to the PATH
Downloading the archive
Writing archive to disk
Extracting the archive
/usr/bin/tar xz -C /__w/cocoa-hep/cocoa-hep/49e15a3c-9cac-4ed7-b481-038727c5650b -f /__w/cocoa-hep/cocoa-hep/49e15a3c-9cac-4ed7-b481-038727c5650b.tar.gz
Resolved version cocoa-hep-cocoa-hep-88d130d
```
Basically the git in CentOS7 container is outdated.
And multiple actions in a row pick this particular commit `88d130d`

So it seems that the actions are brocken. Hard to say what to do. Maybe would make sense to got to CentOS8.

Best regards,

Andrii

**Comment by andriish** [comment]
Well, and yes, 

I would also do 
```
yum -y install libXmu-devel libICE-devel libXext-devel libX11-devel libXt-devel libSM-devel
```
instead of manualy linking the libraries.
